### ISSUE TITLE
Unknown frequency yrPt for fracLut in drq2ins.py

### ISSUE THREAD

**Issue by treerink** [EC-Earth/ece2cmor3#340]
Unknown frequency `yrPt` for variable `fracLut` in `drq2ins.py`:
`````
./drq2ins.py --vars cmip6-data-request/cmip6-data-request-m=CMIP-e=piControl-t=1-p=1/cmvme_CMIP_piControl_1_1.xlsx
`````
gives
`````
 fracLut         Eyr      %               yrPt
 Unknown frequency in LPJ-GUESS Volume estimate for:  fracLut 
`````

**Comment by nierad** [comment]
Hi Thomas, 
Sorry, I don't understand, what needs to be done... Is that something we can fix?

**Comment by nierad** [comment]
Hi again, 

here is a list of unsupported frequencies for lpjg because they are below min. timestep of 1 day.  Is there any way we could adress that?
For "yrPt": By looking at ifs2cmor it seems as if it is treated just like "yr". Is that correct?
If so, we'd treat it just the same.
ERROR:ece2cmor3.lpjg2cmor:The frequency subhrPt for variable fco2antt in table CFsubhr is not supported by lpj2cmor
ERROR:ece2cmor3.lpjg2cmor:The frequency subhrPt for variable fco2nat in table CFsubhr is not supported by lpj2cmor

ERROR:ece2cmor3.lpjg2cmor:The frequency 3hr for variable gpp in table E3hr is not supported by lpj2cmor
ERROR:ece2cmor3.lpjg2cmor:The frequency 3hr for variable ra in table E3hr is not supported by lpj2cmor
ERROR:ece2cmor3.lpjg2cmor:The frequency 3hr for variable rh in table E3hr is not supported by lpj2cmor

EERROR:ece2cmor3.lpjg2cmor:The frequency 3hrPt for variable mrsos in table 3hr is not supported by lpj2cmor
ERROR:ece2cmor3.lpjg2cmor:The frequency 3hr for variable mrro in table 3hr is not supported by lpj2cmor

ERROR:ece2cmor3.lpjg2cmor:The frequency 6hr for variable mrsos in table 6hrPlevPt is not supported by lpj2cmor
ERROR:ece2cmor3.lpjg2cmor:The frequency 6hr for variable tsl in table 6hrPlevPt is not supported by lpj2cmor
ERROR:ece2cmor3.lpjg2cmor:The frequency 6hr for variable mrsol in table 6hrPlevPt is not supported by lpj2cmor

EERROR:ece2cmor3.lpjg2cmor:The frequency yrPt for variable cSoilLut in table Eyr is not supported by lpj2cmor
ERROR:ece2cmor3.lpjg2cmor:The frequency yrPt for variable cVegLut in table Eyr is not supported by lpj2cmor
ERROR:ece2cmor3.lpjg2cmor:The frequency yrPt for variable cLitterLut in table Eyr is not supported by lpj2cmor
ERROR:ece2cmor3.lpjg2cmor:The frequency yrPt for variable cProductLut in table Eyr is not supported by lpj2cmor
ERROR:ece2cmor3.lpjg2cmor:The frequency yrPt for variable cSoil in table Eyr is not supported by lpj2cmor
ERROR:ece2cmor3.lpjg2cmor:The frequency yrPt for variable cVeg in table Eyr is not supported by lpj2cmor

ERROR:ece2cmor3.lpjg2cmor:The frequency yrPt for variable cLitter in table Eyr is not supported by lpj2cmor
ERROR:ece2cmor3.lpjg2cmor:The frequency yrPt for variable cProduct in table Eyr is not supported by lpj2cmor
ERROR:ece2cmor3.lpjg2cmor:The frequency yrPt for variable fracLut in table Eyr is not supported by lpj2cmor

**Comment by goord** [comment]
Hi Lars, I wasn't aware there are yrPt variables requested for IFS, if we treat then as yearly averages then that's certainly incorrect. If lpj-guess really works on a daily timestep, then it may be correct to state that the 3- and 6-hourly fields are simply copies of the midnight value, and the yrPt is just the july 15th value?

**Comment by nierad** [comment]
Hi Gijs,
I don’t know if they are requested for IFS. In cmor_lib.py there  is only one incident of “yrPt” and it looks, as if they are treated equally. And, sorry, that is not IFS, that’s general.

For the subdiurnal freqs: Those are daily values. not midnight, i.e. daily sums.  I think, we can do this. For yrPt I don’t see the point right now.
—

Lars Peter Nieradzik
Institute for Physical Geography and Ecosystem Science
Lund University
Sölvegatan 12
223 62 Lund, Sweden

Phone: +46 (0)46 222 3977

On 14 Dec 2018, at 14:09, Gijs van den Oord <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi Lars, I wasn't aware there are yrPt variables requested for IFS, if we treat then as yearly averages then that's certainly incorrect. If lpj-guess really works on a daily timestep, then it may be correct to state that the 3- and 6-hourly fields are simply copies of the midnight value, and the yrPt is just the july 15th value?

**Comment by treerink** [comment]
Hi Lars,

I think it depends on how you want to treat the yrPt (**P**oint in **t**ime, or instantaneous) variables in LPJ-GUESS. Do you want to label those files just by "yearly" as well? Or do you prefer to label them with "yearly_instantaneous" (or "yearly_Pt"), I don't know if you want to distinguish in LPJ-GUESS. I can imagine you don't care and just let both yearly and yrPt both end up in yearly files, in that case I can quickly fix it. In the other case I can also quickly fix it, then you get for these variables some " yearly_instantaneous" labeled file names  in the instruction file, but LPJ-GUESS has to be able to handle this, i.e. you probably have to add some lines of code.

So what do you prefer:

- option A: all yearly average and instantaneous variables in file all labeled by "yearly"
- option B: the yearly average in files labeled by "yearly" and separately the instantaneous variables in files labeled by "yearly_instantaneous"

**Comment by nierad** [comment]
Thanks Thomas, We'll discuss it. When would the yrPt Point-in-time be? doy 183 (early July)?

**Comment by treerink** [comment]
I added already in comment in `drq2ins.py` "Option A" which also can easily be changed to "Option B".

**Comment by treerink** [comment]
Following the decision above: LPJ-GUESS will not provide yrPt fields. Implemented in 98a108a.

**Comment by treerink** [comment]
Because the 3 and 6 hourly issue is now addressed in #349, this one can be closed.
